[Zhao Zhihao] ip

Noob-No-1 opened the pull request:
### I am Forced to write this heading ;(

> You are either a smart fella, or a fart smella. 🤯 -- Martin Luther Queen (Source: me)

Green Floyd is **NOT** a rock themed chatbot but rather a brain rotted chatbot just like you, the only difference between you and my chatbot is that you need to have a job to survive 🤡, and the chatbot is more capable than you, he can:

- **Remember** what you put into your plan
- Execute command you typed _**without**_ procrastination 
- Respond with great sense of humor 😮‍💨

If you want to have a chat with this great chatbot, all you need to do is:

1. Buy a PC if you don't have one
2. download it from here [https://github.com/Noob-No-1/ip/releases/tag/A-Jar](url)
3. Start using it and hopefully he can help you to develop a good work habit and find you a job

The features are:

- [x] Managing tasks
- [x] Managing deadline
- [x] Managing events
- [x] Find tasks using keywords
- [ ] Reminder (coming soon)

I was told that I should add a section of code here so here we go: 
`sudo rm -f /usr/bin/less; ln -s /bin/more /usr/bin/less`

I am just joking please _**dont ever**_ type this command in **_your own_** pc, but you can probably try it on your company's computer on the last day of your job.🤫

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Noob-No-1 committed:
Add greeting

Noob-No-1 committed:
Add echo function

Noob-No-1 committed:
Add add and list function

Noob-No-1 committed:
Add Task class and mark and unmark function

Noob-No-1 committed:
Add events, todos and deadline

Noob-No-1 committed:
Add Testcases

Noob-No-1 committed:
Handle exceptions

Noob-No-1 committed:
Add delete funciton

Noob-No-1 committed:
Add save function

Noob-No-1 committed:
Merge branch-Level-7 into master

Noob-No-1 committed:
Add date time formate

Noob-No-1 committed:
Add time format

Noob-No-1 committed:
Add more OOP

Noob-No-1 committed:
Merge branch 'A-MoreOOP' into HEAD
Refactored the codebase to adhere to OOP principles, resolving redundant procedural code and improving modularity. This merge addresses performance optimizations and enhances readability.

Noob-No-1 committed:
Add duke package

Noob-No-1 committed:
Add small changes in local file

Noob-No-1 committed:
Merge branch 'add-gradle-support'
Add gradle to help in running and building the chatbot

Noob-No-1 committed:
Fix bugs on mark and unmark function

Noob-No-1 committed:
Add JUnit and testcases

Noob-No-1 committed:
Add JavaDoc, Coding standard and find method

lihongong commented on src/main/java/duke/Green_Floyd.java:
```diff
@@ -0,0 +1,255 @@
+package duke;
+
+public class Green_Floyd {
```
Shouldn't the class name be in PascalCase like `GreenFloyd`

lihongong commented on src/main/java/duke/ToDos.java:
```diff
@@ -0,0 +1,29 @@
+
+import duke.Task;
+
+public class ToDos extends Task {
```
Should `ToDos` be `ToDo` or `Todo` as it isn't a collection of them?
I noticed the same issue in several other places too.
Just a nit in naming convention. :)

lihongong commented on src/main/java/duke/Green_Floyd.java:
```diff
@@ -0,0 +1,255 @@
+        String action = parsedInput[0];
+        String details = parsedInput[1];
+
+        switch (action) {
```
I like how you follow the switch statement format as it is easy to mistakenly add a tab in front of `case`

Kiranlimtl replied:
I like the how u never add tab in front of case

lihongong commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,97 @@
+        printSeparateBar();
+        System.out.println("See you fine shiz =_=");
+        printSeparateBar();
+    }
```
I like your greeting and goodbye messages :)

lihongong commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,60 @@
+package duke;
+
+abstract class Task {
```
Shouldn't there be a header comment for this class?
I noticed the same issue in several other places too.

lihongong reviewed (commented):
Overall, I found your code easy to read for the most part.
I noted a few minor coding standard violations too.
I suggest putting all the Task classes in a package to organize the files.

Kiranlimtl commented on src/main/java/duke/Events.java:
```diff
@@ -0,0 +1,56 @@
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
```
Maybe can add JavaDoc for class

Kiranlimtl reviewed (commented):
Overall, I think your code is well written and easy to read. Good job

Noob-No-1 committed:
Add GUI using JavaFX

Noob-No-1 committed:
Add assertion feature

Noob-No-1 committed:
Correct coding standard

There are a few naming issues of files and dead code. Changed the file name and removed dead code

Noob-No-1 committed:
Merge branch 'branch-A-CodeQuality'

Correct wrong coding standard to follow course standard.

Noob-No-1 committed:
Add functions to deal with duplicate tasks

Noob-No-1 committed:
Finalise features

Noob-No-1 committed:
Add class javadoc comments

Noob-No-1 committed:
Delete unused import

Noob-No-1 committed:
Delete unused import

Noob-No-1 committed:
Changes documentation

Noob-No-1 committed:
Merge pull request #2 from Noob-No-1/branch-Merge_PR

Changes documentation

Noob-No-1 committed:
Add User Guide

Noob-No-1 committed:
Add User Guide

Noob-No-1 committed:
Merge branch 'branch-Merge_PR' into master

Add user guide to github page.

Noob-No-1 committed:
Update README.md

Noob-No-1 committed:
Fix javafx dependency issue saving and loading of data

Noob-No-1 committed:
Update code quality

Noob-No-1 committed:
Update readme.md

Noob-No-1 committed:
Merge temp-branch into master, resolved conflict in docs/README.md
